jerry-ext/jerry-module/include/jerry-module.h

```diff
@@ -0,0 +1,100 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_MODULE_H
+#define JERRYX_MODULE_H
+
+#include "jerryscript.h"
+
+#ifdef __cplusplus
+#define EXTERN_C_START extern "C" {
+#define EXTERN_C_END }
+#else
+#define EXTERN_C_START
+#define EXTERN_C_END
+#endif
+
+#define JERRYX_C_CTOR(fn)                          \
+static void fn(void) __attribute__((constructor)); \
+static void fn(void)
+
+#define JERRYX_MODULE_VERSION 1
+
+typedef struct jerryx_module_link
+{
+  jerry_char_t *name;
+  struct jerryx_module_link *next;
+} jerryx_module_header_t;
```

Comment:
I'd try to move the link list implementation details out of the .h and into the .c, to make it easier to add alternative implementations in the future.